src/sqlfluff/dialects/dialect_ansi.py

```diff
@@ -943,6 +944,13 @@ class QualifiedNumericLiteralSegment(BaseSegment):
         Ref("ExpressionSegment"),
         # A Cast-like function
         Sequence(Ref("ExpressionSegment"), "AS", Ref("DatatypeSegment")),
+        # Trim function
+        Sequence(
```

Comment:
Are we checking the function name, or can this be used with any function? Seems like the latter. Any concerns with allowing that?

Reply:
Yeah I thought about that. You're right we do allow with any function. But very specific set of syntax and grammar so think will only match that.

This who `FunctionContentsGrammar` is made up of lots of syntax types that could be specific to certain function names but isn't, and this PR continues that theme.

To limit it to certain function names would require a change similar to #3004 where we get around it being a keyword (so it's still a function name type). But that gets messy and potentially unmaintainable pretty quickly. Hence I did initially implement #3004 in a similar manner to this. However, as that was specific to BigQuery, and similar to another pattern in Function I changed my mind on that.

For this TRIM one, I think we follow the previous pattern, as more generic and unlikely to match incorrectly.